internal/provider/waypoint/resource_waypoint_add_on_definition.go

```diff
@@ -252,6 +256,18 @@ func (r *AddOnDefinitionResource) Create(ctx context.Context, req resource.Creat
 		})
 	}
 
+	tfProjID := plan.TerraformProjectID.ValueString()
+	tfWsName := plan.TerraformCloudWorkspace.Name.ValueString()
+	if tfWsName == "" {
+		// NOTE: this field is optional anyways, so if its unset, lets just use
+		// the template name for it.
+		tfWsName = plan.Name.ValueString()
+	}
+	tfWsDetails := &waypointModels.HashicorpCloudWaypointTerraformCloudWorkspaceDetails{
+		Name:      tfWsName,
+		ProjectID: tfProjID,
+	}
+
```

Comment:
If the `Name` is optional, what do you think of just doing away with it? Because `terraform_cloud_workspace_details` is currently `Optional`, but if users do specify it they need to specify both `name` and `terraform_project_id`. The code above only uses the `name` if it's specified in `terraform_cloud_workspace_details` (if I read it correctly).

Right now the only way it's used is if it's populated in the block, so the config will look like this:

```hcl
  terraform_project_id = "<ID for project foo>"
  terraform_cloud_workspace_details = {
    name                 = "<Name for project foo>"
    terraform_project_id = "<ID for project foo>"
  }
```

Maybe we should do away with the block entirely and just have this:

```hcl
terraform_project_id = "<ID for project foo>"
```

What do you think?

Reply:
Hrm, I though I made this change such that `terraform_cloud_workspace_details` is completely optional now to define in a module: https://github.com/hashicorp/terraform-provider-hcp/pull/1052/files#diff-514a4247ea434f616ba275ad89920b100f00d291f761cf4948bf3bab5da58283R116-R117

I wanted to remove this too, but I think the Waypoint API requires that this struct is set in a request. And I didn't want to break that.

Reply:
It sounds like we are going to mark `terraform_cloud_workspace_details` as deprecated via `DeprecationMessage`. Then let users just set `terraform_project_id`, and construct this behind the scenes to satisfy the Waypoint API validation.